docs/core/testing/mstest-analyzers/mstest0045.md

```diff
@@ -30,6 +30,11 @@ A test method uses <xref:Microsoft.VisualStudio.TestTools.UnitTesting.TimeoutAtt
 
 ## Rule description
 
+There is no way to gracefully abort a running thread. So, for a timeout to work, there are two ways:
+
+1. Stop observing the thread running the test. But this can be problematic in many cases and might make the remaining tests unstable due to potential race conditions. This is non-cooperative cancellation.
+2. A cancellation is requested once the timeout is reached, and it's the test responsibility to terminate on request. This is cooperative cancellation.
```

Comment:
```suggestion
- Stop observing the thread running the test. But this can be problematic in many cases and might make the remaining tests unstable due to potential race conditions. This is *non-cooperative cancellation*.
- Request cancellation once the timeout is reached, and it's the test responsibility to terminate on request. This is *cooperative cancellation*.
```